src/osdc/Filer.cc

```diff
@@ -339,20 +339,27 @@ struct C_PurgeRange : public Context {
   PurgeRange *pr;
   C_PurgeRange(Filer *f, PurgeRange *p) : filer(f), pr(p) {}
   void finish(int r) override {
-    filer->_do_purge_range(pr, 1);
+    filer->_do_purge_range(pr, 1, r);
   }
 };
 
-void Filer::_do_purge_range(PurgeRange *pr, int fin)
+void Filer::_do_purge_range(PurgeRange *pr, int fin, int err)
 {
   PurgeRange::unique_lock prl(pr->lock);
+  if (err != 0) {
+    pr->oncommit->complete(err);
+    prl.unlock();
+    delete pr;
+    return;
+  }
```

Comment:
I think we should remove this code block

Reply:
Done

Reply:
you removed all changes in Filer

Reply:
Sorry. I have added them back.

Reply:
need code save error code

Reply:
I have saved the error code in `PurgeRange`.  But we still need to handle the error in `_do_purge_range`, right? If yes then what is wrong with the handling done in earlier patch.

Reply:
the origin code stops truncate when it gets -ENOENT. -ENOENT is not a error because there might be some holes in file range.

Reply:
Thanks @ukernel for explaining. If we get other errors then shouldn't it stop immediately ?

Reply:
I prefer not to do stop